### DISCUSSION TITLE
[V5] Error: Cannot define "transactionListener" on model, since it is not defined as a model property

### DISCUSSION THREAD

**Discussion by HemanthKodhanda** [adonisjs/core#2100] [Help]
Hi , I am getting below error 
**Error: Cannot define "transactionListener" on "Personnel" model, since it is not defined as a model property** , with await **user.related('personnel').save(personnel)**

user model -> hasone relation with personnel model.  please let me know what is the issue.

**Comment by thetutlage** [comment]
Hard to comment without looking at any code especially the models and the code that invokes the `save` method

**Reply by genu** [reply]
@thetutlage 

Seeing this too.  The context that I see this happening in my case is when you try to use a `Factory` together with a `Seeder`.

For example,

```ts
export default class UserSeeder extends BaseSeeder {
  public async run() {
    const users = await UserFactory.makeMany(10)

    await User.createMany(users)
  }
}
```

Will throw this exception:

```
❯ error     database/seeders/User
  Cannot define "transactionListener" on "User" model, since it is not defined as a model property
```


From my limited knowledge of Adonis (started using it today), is that the Model's `createMany` expects `POJOs`.

If that's the case, a note in the docs might be helpful, or some kind of indication how Factories and Seeders should be used together (or not). My gut inclination was to use Factories when seeding.

**Reply by thetutlage** [reply]
`UserFactory.makeMany` returns an array of model instances. I think, the Typescript static types can help here. 

And yes, you are create `createMany` expects an array of plain objects and not model instances. 

You can just persist the models directly using the `createMany` method on the factory

```ts
const users = await UserFactory.createMany(10)
console.log(users)
```
